lib/hydra/works.rb

```diff
@@ -64,21 +64,38 @@ module Works
     autoload :GenerateThumbnail,                   'hydra/works/services/generic_file/generate/thumbnail'
     autoload :PersistDerivative,                   'hydra/works/services/generic_file/persist_derivative'
 
-    # model validations
-    def self.collection? collection
+    # model validations 
+    def self.works_collection? collection
```

Comment:
Esme,

Once I remove these methods from hydra/works/works.rb

works_collection?
works_generic_file?
works_generic_work?

because they live in hydra/works/model/concerns/generic_file_behavior.rb

What do I do about spec/hydra/work_spec.rb  ?

The test that ask for

works_collection?
works_generic_file?
works_generic_work?

come back as undefined.



On Mon, Aug 3, 2015 at 7:25 AM, Esmé Cowles <notifications@github.com>
wrote:

> In lib/hydra/works.rb
> <https://github.com/projecthydra-labs/hydra-works/pull/182#discussion_r36076252>
> :
>
> > @@ -64,21 +64,38 @@ module Works
> >      autoload :GenerateThumbnail,                   'hydra/works/services/generic_file/generate/thumbnail'
> >      autoload :PersistDerivative,                   'hydra/works/services/generic_file/persist_derivative'
> >
> > -    # model validations
> > -    def self.collection? collection
> > +    # model validations
> > +    def self.works_collection? collection
>
> These methods should be removed -- these same methods are implemented in
> models/concerns/work_behavior.rb, e.g.: #works_collection
> <https://github.com/blancoj/hydra-works/blob/issue_164/lib/hydra/works/models/concerns/generic_work_behavior.rb#L34>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projecthydra-labs/hydra-works/pull/182/files#r36076252>
> .
>